File: mod/library/server/handle-message/exporter/process_enhancement.lua
local get_desc_from_table = SMODS.load_file("mod/server/handle-message/exporter/_common/get_desc_from_table.lua")()
local check_for_override = SMODS.load_file("mod/server/handle-message/exporter/_common/check_for_override.lua")()
local output_image = SMODS.load_file("mod/server/handle-message/exporter/_common/output_image.lua")()

local function process_enhancement(sets, card)
	local item = {}
	local center = card.config.center
	output_image(center)
	if card.ability_UIBox_table then
		item.name = localize { type = 'name_text', key = center.key, set = center.set }
		item.description = get_desc_from_table(card.ability_UIBox_table.main)
	end
	item.key = center.key
	item.set = center.set
	if center.mod and center.mod.id ~= "Aura" and center.mod.id ~= "aure_spectral" then
		item.mod = center.mod.id
	end
	item.tags = {}
	item.image_url = check_for_override("images/" .. center.key:gsub("?", "_") .. ".png")
	if item.name then
		sets[item.set][item.key] = item
	end
end

return process_enhancement

File: mod/library/server/handle-message/exporter/process-enhancement.ts
import {
	getDescriptionFromTable,
	outputImage
} from "./_common/_exports.ts";

/**
 *
 * @param card
 * @noSelf
 * @example
 */
const processEnhancement = (card: typeof CardType) => {
	const item: any = {};
	const { center } = card.config;

	if (center.atlas === undefined) {
		center.atlas = "centers";
	}

	outputImage(center);

	if (card.ability_UIBox_table) {
		item.name = localize({
			key: center.key,
			set: center.set,
			type: "name_text"
		});
		item.description = getDescriptionFromTable(card.ability_UIBox_table.main);
	}

	item.key = center.key;
	item.set = center.set;

	if (center.mod !== undefined && center.mod.id !== "Aura" && center.mod.id !== "aure_spectral") {
		item.mod = center.mod.id;
	}

	item.tags = {};

	item.image_url = `${center.key.replaceAll("?", "_")}.png`;

	if (item.name !== undefined) {
		return item;
	}
};

export default processEnhancement;
